dali/pipeline/operators/util/lookup_table.cc

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/pipeline/operators/util/lookup_table.h"
+
+namespace dali {
+
+constexpr auto kMaxKey = LookupTable<CPUBackend>::kLookupTableSize - 1;
+
+template<>
+void LookupTable<CPUBackend>::RunImpl(SampleWorkspace &ws) {
+  const auto &input = ws.Input<CPUBackend>(0);
+  auto &output = ws.Output<CPUBackend>(0);
+  auto data_size = input.size();
+  TYPE_SWITCH(input.type().id(), dali::type2id, InputType, (uint8_t, int16_t, int32_t, uint64_t), (
+    TYPE_SWITCH(output_type_, dali::type2id, OutputType, (float, uint8_t, int16_t, int32_t), (
+      constexpr bool check_range =
```

Comment:
Please add a comment that we do not check range in those cases because the range is smaller than the biggest possible index.
Also, it would be better to use
```suggestion
      constexpr bool check_range = std::numeric_limits<InputType>::max() > kMaxKey;
```

Reply:
I had that before but then I thought that I might want to add float16 at some point